scripts/dts/new_extract.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import sys
+
+import edtlib
+
+
+def str2ident(s):
```

Comment:
One of the most confusing things when trying to understand the extractor is to find things like:

`DT_SW0_GPIO_CONTROLLER`

and not know where this came from because `git grep GPIO_CONTROLLER` gives you nothing. I'd be very grateful if you could document this somewhere (the construction of the macros from the DT labels/identifiers) so that in the future this is easy to understand by everybody.

Reply:
> Where? I can't seem to find them

In the work branch at https://github.com/ulfalizer/zephyr/tree/edtlib-wip.

> As discussed offline, this should all be licensed under Apache 2.0 and attributed to Nordic Semiconductor ASA

Done

> One of the most confusing things when trying to understand the extractor is to find things like:
> 
> `DT_SW0_GPIO_CONTROLLER`
> 
> and not know where this came from because `git grep GPIO_CONTROLLER` gives you nothing. I'd be very grateful if you could document this somewhere (the construction of the macros from the DT labels/identifiers) so that in the future this is easy to understand by everybody.

Yup, the DeviceTree stuff is lacking a lot of documentation. Not sure I can do it all on my own though, because I'm not familiar with the motivation behind all of those names and their components.

Documentation should definitely be a focus later at least. As it is now, almost everything has to be reverse-engineered. You should be able to go somewhere and find "if the .dts file has this, you'll get this define".